### PR TITLE
PHP7 compatibility by @timmipetit

### PR DESCRIPTION
**Pull request by zkiiito**
I've merged the modifications by @timmipetit from the Recras fork, and here is it as a pull request. Still php 5.3 compatible.

### CODE REVIEW

**Comment by j0k3r** [comment]
Oh thanks !

This'll fix https://github.com/LExpress/symfony1/issues/118

File: .travis.yml [hunk 1]
```diff
@@ -12,7 +12,6 @@ php:
 matrix:
     fast_finish: true
     allow_failures:
-        - php: 7.0
```
Comment (j0k3r) [inline .travis.yml]: Could you add _nightly_ and allow it to fail? So the build will run on PHP 7.1.

**Comment by j0k3r** [comment]
Could you squash your latest two commits? Hopefully it'll trigger Travis again ...

**Comment by zkiiito** [comment]
well, the readme also needed an update, and travis is running

**Comment by miniche** [comment]
Thanks a lot! 👍 

Have you tried these changes on a real project with other libraries like Doctrine1? [Sadly, the last build seems to be broken on PHP 7](https://travis-ci.org/LExpress/doctrine1/jobs/118475181)

**Comment by j0k3r** [comment]
:arrow_right: https://github.com/LExpress/doctrine1/pull/17

**Comment by zkiiito** [comment]
we're migrating our codebase from 5.3 to be compatible with 7.0 (and if possible, hhvm too), so I can say that on 5.3 this works, we'll test on 7.0 soon.
But, I think the Recras guys are using it in prod, basicly everything important here were done by @timmipetit

**Comment by timmipetit** [comment]
Yeah, we've been running this in production for a couple of months now.

**Comment by j0k3r** [comment]
@timmipetit are you also running doctrine1 as well?

**Comment by timmipetit** [comment]
Yes, also based on our own fork (also with php7 fixes): https://github.com/Recras/doctrine1

**Comment by miniche** [comment]
Ok for Doctrine1. 

@j0k3r: Do we wait to merge doctrine1 PR before this one?

**Comment by j0k3r** [comment]
We can merge both of them and tag them later.
Hopefully some user still use `dev-master` in their `composer.json` :)

**Comment by miniche** [comment]
Alright. Could be merged ASAP if everyone is OK :)

**Pull request merged**